app/src/main/java/com/celzero/bravedns/ui/SettingsFragment.kt

```diff
@@ -872,16 +952,16 @@ class SettingsFragment : Fragment(R.layout.activity_settings_screen) {
         b.settingsActivityOnDeviceLastUpdatedTimeTxt.visibility = View.VISIBLE
         val timeStamp = persistentState.localBlockListDownloadTime
         b.settingsActivityOnDeviceLastUpdatedTimeTxt.text = getString(R.string.settings_local_blocklist_version, Utilities.convertLongToDate(timeStamp))
-        val path: String = requireContext().filesDir.canonicalPath + "/" +timeStamp
+        val path: String = requireContext().filesDir.canonicalPath + "/" + timeStamp
         if (appMode?.getBraveDNS() == null) {
             GlobalScope.launch(Dispatchers.IO) {
                 try {
-                    if (DEBUG) Log.d(LOG_TAG, "$FILETAG Local brave dns set call from settings fragment newBraveDNSLocal : $path")
+                    if (DEBUG) Log.d(LOG_TAG_VPN, "$FILETAG Local brave dns set call from settings fragment newBraveDNSLocal : $path")
                     val braveDNS = Dnsx.newBraveDNSLocal(path + Constants.FILE_TD_FILE, path + Constants.FILE_RD_FILE, path + Constants.FILE_BASIC_CONFIG, path + Constants.FILE_TAG_NAME)
                     appMode?.setBraveDNSMode(braveDNS)
                     persistentState.localBlocklistEnabled = true
-                }catch (e: Exception){
-                    Log.w(LOG_TAG, "Exception while setting blocklist: ${e.message}", e)
+                } catch (e: Exception) {
+                    Log.w(LOG_TAG_VPN, "Exception while setting blocklist: ${e.message}", e)
                     persistentState.localBlocklistEnabled = false
                     if (isAdded) {
                         (context as HomeScreenActivity).runOnUiThread {
```

Comment:
move `runOnUiThread` into the `showFileCorruptionDialog` (rename).

Also, does `updateDownloadFailure` require ui-thread? If not, see how doing the above avoids such bugs.

Reply:
Renamed to `showFileCorruptionDialog`, `updateDownloadFailure()` - Updates the UI elements.
Moved runOnUiThread to method.